server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/SetMailboxesMethodTest.java

```diff
@@ -366,6 +368,110 @@ public void subscriptionUserShouldBeChangedWhenUpdateMailbox() throws Exception
+            .body(requestBody)
+            .post("/jmap");
+
+        assertThat(mailboxProbe.listSubscriptions(username.asString()))
+            .contains("root.second.thirdtest.fourth")
+            .doesNotContain("root.second.third.fourth");
+    }
+
+    @Test
+    public void subscriptionUserShouldBeChangedForAllChildrenWhenUpdateParentMailbox() throws Exception {
+        MailboxId parentId = mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, username.asString(), "root");
+
+        String secondMailboxName = "second";
+        MailboxId secondMailboxId = createSubMailBox(parentId, secondMailboxName);
+
+        String thirdMailBoxName = "third";
+        MailboxId thirdMailboxId = createSubMailBox(secondMailboxId, thirdMailBoxName);
+
+        String fourthMailboxName = "fourth";
+        createSubMailBox(thirdMailboxId, fourthMailboxName);
+
+        String requestBody =
+            "[" +
+                "  [ \"setMailboxes\"," +
+                "    {" +
+                "      \"update\": {" +
+                "        \"" + secondMailboxId.serialize() + "\" : {" +
+                "          \"name\" : \"secondtest\" ,"  +
+                "          \"parentId\" : \"" + parentId.serialize() + "\""  +
+                "        }" +
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]";
+        with()
+            .header("Authorization", accessToken.asString())
+            .body(requestBody)
+            .post("/jmap");
+
+        assertThat(mailboxProbe.listSubscriptions(username.asString()))
+            .contains("root.secondtest.third.fourth")
+            .doesNotContain("root.second.third.fourth");
+    }
+
+    private MailboxId createSubMailBox(MailboxId parentMailboxId, String childMailboxName) {
+        String createChildMailbox =
+            "[" +
+                "  [ \"setMailboxes\"," +
+                "    {" +
+                "      \"create\": {" +
+                "        \"whatever\" : {" +
+                "          \"name\" : \"" + childMailboxName + "\"," +
+                "          \"parentId\" : \"" + parentMailboxId.serialize() + "\"" +
+                "        }" +
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]";
+
+        Response response = given()
+            .header("Authorization", accessToken.asString())
+            .body(createChildMailbox)
+        .when()
+            .post("/jmap")
+        .thenReturn();
+
+        String responseStr = response.body().asString();
+        responseStr = responseStr.substring(responseStr.indexOf("created\":"));
+        responseStr = responseStr.substring(responseStr.indexOf("id\":") + 4);
+        String mailboxId = responseStr.substring(0, responseStr.indexOf(",")).replaceAll("\"", "");
+        return new TestId.Factory().fromString(mailboxId);
```

Comment:
I'm unsure you can assume it is always a TestId. This test have to run against the Cassandra backend which obviously do not rely on TestId.

Maybe for the time being we can return a String?

Reply:
It just a strong typing, we passed `String` into `url` call right?

Reply:
Or adding an abstract method that would return the `MailboxId.Factory` that you would implement in different test implementations, to rely on different factories?